File: PostsService/src/models/post.rs

use serde_derive::{Serialize, Deserialize};

use super::comments::Comment;


#[derive(Serialize, Clone, Deserialize)]
pub struct Post {
  pub id: String,
  pub createdAt: String,
  pub text: String,
  pub writtenBy: String,
  pub likedBy: Vec<String>,
  pub comments: Vec<Comment>
}

File: FrontendService/src/plugins/posts/models/Post.ts
export type Post = {

  id: string,
  createdAt: string,
  text: string,
  writtenBy: string

  likedBy: string[],
  comments: Comment[]
}

export type Comment = {
  id: string,
  createdAt: string,
  postId: string,
  userId: string,
  text: string
}